src/main/java/com/example/glovoordermanagementapi/controller/APIController.java

```diff
@@ -1,4 +1,218 @@
 package com.example.glovoordermanagementapi.controller;
 
+import com.example.glovoordermanagementapi.exception.NotFoundException;
+import com.example.glovoordermanagementapi.model.Order;
+import com.example.glovoordermanagementapi.model.Product;
+import com.example.glovoordermanagementapi.service.OrderService;
+import com.example.glovoordermanagementapi.service.ProductService;
+import lombok.Data;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Data
+@Controller
+@RequestMapping("/api/orders")
 public class APIController {
+
+    private static ProductService productService;
+    @Autowired
+    private final OrderService orderService;
+
+    public APIController(OrderService orderService) {
+        this.orderService = orderService;
+    }
+
+    @SneakyThrows
+    @PostMapping("/create")
+    public ResponseEntity<Order> createOrder(@RequestParam int orderId) {
+        if (orderId <= 0) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        // Створення об'єкта Order
+        Order orderNew = orderService.createOrder(orderId);
+
+        // Виклик сервісу для створення замовлення
+        Order createdOrder = orderService.createOrder(orderNew.getId());
+
+        return new ResponseEntity<>(createdOrder, HttpStatus.CREATED);
+    }
+
+    @GetMapping("/{orderId}")
+    public ResponseEntity<Order> getOrderById(@PathVariable int orderId) throws NotFoundException {
+        ResponseEntity<Order> result;
+        Order order = orderService.getOrderById(orderId);
+        if (order != null) {
+            result = new ResponseEntity<>(order, HttpStatus.OK);
+        } else {
+            result = new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return result;
+    }
+
+    @PutMapping("/{orderId}")
+    public ResponseEntity<Void> updateOrder(@PathVariable int orderId, @RequestBody Order updatedOrder)
+            throws NotFoundException {
+        Order existingOrder = orderService.getOrderById(orderId);
+
+        if (existingOrder != null) {
+
+            existingOrder.setProducts(updatedOrder.getProducts());
+
+
+            orderService.updateOrder(existingOrder);
+
+            return new ResponseEntity<>(HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @GetMapping("/displayAllInfo")
+    public ResponseEntity<Void> displayAllOrdersInfo() {
+        orderService.displayAllOrdersInfo();
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RestController
+    @RequestMapping("/api/orders")
+    public static class OrderController {
+
+        private final OrderService orderService;
+
+        @Autowired
```

Comment:
As i remember Spring will automatically inject the OrderService without using @Autowired